cppwinrt/helpers.h

```diff
@@ -498,6 +498,27 @@ namespace cppwinrt
         return result;
     }
 
+    static bool implements_interface(TypeDef const& type, std::string_view const& name)
+    {
+        for (auto&& impl : type.InterfaceImpl())
+        {
+            const auto iface = impl.Interface();
+            if (iface.type() != TypeDefOrRef::TypeSpec && type_name(iface) == name)
+            {
+                return true;
+            }
+        }
+
+        if (auto base = get_base_class(type))
```

Comment:
There can be a hierarchy of bases...

Reply:
But this can get expensive (e.g. how long cppwinrt takes to run). Will the formatter not work polymorphically, with anything that is convertible to `IStringable` rather than specific types (like `std::hash`)? I never liked that about hash...

Reply:
Why not just enable it for `hstring` and `IStringable` for now and developers can always cast if needed. The idea was to eventually remove everything inside `WINRT_LEAN_AND_MEAN` in a future (major) version anyway.

Reply:
> There can be a hierarchy of bases...

This is a recursive call, so it will check each base until it hits the end of the hierarchy. Or is there anything else I'm missing?

> But this can get expensive (e.g. how long cppwinrt takes to run).

I haven't personally seen much impact on cppwinrt's runtime when developing this. I can do measurements if wanted.

> Will the formatter not work polymorphically, with anything that is convertible to IStringable rather than specific types (like std::hash)? I never liked that about hash...

Yeah unfortunately std::formatter behaves exactly like std::hash there. But you can "enable" the desired behavior just by inheritance like I do in this PR. Since then the compiler can figure out the implicit conversion at the call site.

> Why not just enable it for hstring and IStringable for now and developers can always cast if needed.

I think this will lead to confusion, especially since the error message that would be given when called on a type that implements IStringable is deep within std::format internals and can't really give anything useful, but then it works when called on a direct IStringable. Giving a good error message would require similar codegen to what we're already doing, so why not just enable the support directly.

> The idea was to eventually remove everything inside WINRT_LEAN_AND_MEAN in a future (major) version anyway.

Seeing as there's only like a dozen of types in the system APIs that implement IStringable, we could maybe remove this from WINRT_LEAN_AND_MEAN already and enable it all the time.

Reply:
Yes, if there's a small number of types rather just always enable - if its useful it shouldn't be in `WINRT_LEAN_AND_MEAN`.